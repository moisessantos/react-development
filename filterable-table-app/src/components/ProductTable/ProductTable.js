import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ProductRow, ProductCategoryRow } from '../';

class ProductTable extends Component {
  render() {
    const rows = [];
    let lastCategory = null;
    
    this.props.products.forEach((product) => {
      if (product.category !== lastCategory) {
        rows.push(
          <ProductCategoryRow
            category={product.category}
            key={product.category} />
        );
      }
      rows.push(
        <ProductRow
          product={product}
          key={product.name} />
      );
      lastCategory = product.category;
    });

    return (
      <table className='table container'>
        <thead>
          <tr className='header'>
            <th>Name</th>
            <th>Price</th>
          </tr>
        </thead>
        <tbody>{rows}</tbody>
      </table>
    );
  }
}

ProductTable.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape({
    category: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    stocked: PropTypes.bool,
    name: PropTypes.string.isRequired,
  })).isRequired,
};
export default ProductTable;
