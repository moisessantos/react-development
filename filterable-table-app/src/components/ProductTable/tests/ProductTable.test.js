import React from 'react';
import { shallow } from 'enzyme';
import ProductTable from '..';

describe('ProductTable component', () => {
  it('should render snapshot', () => {
    const wrapper = shallow(<ProductTable products={[
      {category: 'Sporting Goods', price: '$49.99', stocked: true, name: 'Football'},
      {category: 'Sporting Goods', price: '$29.99', stocked: false, name: 'Basketball'},
      {category: 'Electronics', price: '$99.99', stocked: true, name: 'iPod Touch'}
    ] } />);
    expect(wrapper).toMatchSnapshot();
  });
});
