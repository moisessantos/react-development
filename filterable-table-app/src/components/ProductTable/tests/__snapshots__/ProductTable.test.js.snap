// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTable component should render snapshot 1`] = `
<table
  className="table container"
>
  <thead>
    <tr
      className="header"
    >
      <th>
        Name
      </th>
      <th>
        Price
      </th>
    </tr>
  </thead>
  <tbody>
    <Memo(ProductCategoryRow)
      category="Sporting Goods"
      key="Sporting Goods"
    />
    <Memo(ProductRow)
      key="Football"
      product={
        Object {
          "category": "Sporting Goods",
          "name": "Football",
          "price": "$49.99",
          "stocked": true,
        }
      }
    />
    <Memo(ProductRow)
      key="Basketball"
      product={
        Object {
          "category": "Sporting Goods",
          "name": "Basketball",
          "price": "$29.99",
          "stocked": false,
        }
      }
    />
    <Memo(ProductCategoryRow)
      category="Electronics"
      key="Electronics"
    />
    <Memo(ProductRow)
      key="iPod Touch"
      product={
        Object {
          "category": "Electronics",
          "name": "iPod Touch",
          "price": "$99.99",
          "stocked": true,
        }
      }
    />
  </tbody>
</table>
`;
